network/nat/plugin.go

```diff
@@ -86,18 +86,28 @@ func (p *plugin) Startup(n *network.Network) {
 	log.Info().
 		Int("internal_port", p.internalPort).
 		Int("external_port", p.externalPort).
-		Msgf("external port now forwards to your local port")
+		Msg("external port now forwards to your local port")
 
 	p.gateway = gateway
 
 	info.Host = p.externalIP.String()
 	info.Port = uint16(p.externalPort)
 
+	n.InternalPort = &p.internalPort
+
 	// Set peer information based off of port mapping info.
 	n.Address = info.String()
 	n.ID = peer.CreateID(n.Address, n.GetKeys().PublicKey)
 
-	log.Info().Msgf("other peers may connect to you through the address %s.", n.Address)
+	go func() {
+		for {
+			time.Sleep(60 * time.Second)
+			_, err = gateway.AddPortMapping("tcp", p.internalPort, "noise", 60*time.Second)
```

Comment:
Thanks for your contribution. When have you noticed the port dropping? Is it replicable?

Reply:
When connecting via my router, connections were refused ~50% of the time with a 1s timeout. We should quantifiable measure whether we need to ping the router but qualitatively I had a smoother experience with connecting/disconnecting.

Reply:
If you are referring to the local port, it is not being correctly listened to... Use upnpc and nmap to verify

Reply:
@jack0, I pushed a new commit that removes this. Please let the NAT traversal fix through.